checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -322,7 +322,7 @@ private void refineNeqSequenceLength(
                             atypeFactory, (FieldAccessNode) lengthAccess);
             receiver = fa.getReceiver();
 
-        } else if (IndexUtil.isStringLengthInvocation(lengthAccess, atypeFactory.imf)) {
+        } else if (atypeFactory.imf.isStringLengthInvocation(lengthAccess)) {
```

Comment:
I made a similar comment on panacekcz#6, but this looks like a code smell because you're reading through a field. You should add a getter to the type factory that just returns the Index Method Identifier and use it here.